server.go

```diff
@@ -664,215 +538,161 @@ func (s *Server) handlerMovie(w http.ResponseWriter, r *http.Request) {
-						}
-					} else {
-						s.l.Error("getTitle(): " + err.Error())
-					}
-
-					results, err = getMovieData(&sourceAPI)
-
-					if err != nil {
-						// error while getting data from the api
-						errors = append(errors, err.Error())
-					}
-				} else {
-					errors = append(errors, "Jikan API usage was not enabled by the site administrator")
-					rerenderSite = true
-					return nil, errors, rerenderSite
+		if exists {
+			s.l.Debug("Movie already exists")
+			data.ErrorMessage = append(data.ErrorMessage, "Movie already exists in database")
+			data.ErrAutofill = true
+			return nil, nil
+		}
 
-				}
+		results = append(results, remarkstext)
+		return results, links
 
-			} else if strings.Contains(sourcelink, "imdb") {
+	} else if strings.Contains(sourcelink, "imdb") {
+		s.l.Debug("IMDB link")
 
-				tmdbEnabled, err := s.data.GetCfgBool("TmdbEnabled", DefaultTmdbEnabled)
-				if err != nil {
-					s.l.Error("Error while retriving config value 'TmdbEnabled':\n %v", err)
-					errors = append(errors, "Something went wrong :C")
-					rerenderSite = true
-					return nil, errors, rerenderSite
-				}
+		results, err = s.handleTmdb(data, w, r, sourcelink)
 
-				if tmdbEnabled {
-					// Retrieve token from database
-					token, err := s.data.GetCfgString("TmdbToken", "")
-					if err != nil || token == "" {
-						errors = append(errors, "TmdbToken is either empty or not set in the admin config")
-						rerenderSite = true
-						return nil, errors, rerenderSite
-					}
-
-					// get the movie id
-					rgx := regexp.MustCompile(`[htp]{4}s?:\/\/[^\/]*\/title\/(tt[0-9]*)`)
-					match := rgx.FindStringSubmatch(sourcelink)
-					var id string
-					if len(match) < 2 {
-						errors = append(errors, "Could not retrive movie id from provided link.")
-						rerenderSite = true
-						return nil, errors, rerenderSite
-					} else {
-						id = match[1]
-					}
-
-					sourceAPI := tmdb{id: id, token: token, l: s.l}
-
-					// Return early when the title already exists
-					err = sourceAPI.requestResults()
-					if err != nil {
-						errors = append(errors, err.Error())
-						rerenderSite = true
-						return nil, errors, rerenderSite
-					}
-
-					title, err := sourceAPI.getTitle()
-					if err == nil {
-						exists, _ := s.data.CheckMovieExists(title)
-						if err == nil {
-							if exists {
-								errors = append(errors, "Movie already exists")
-								rerenderSite = true
-								return nil, errors, rerenderSite
-							}
-						}
-
-						results, err = getMovieData(&sourceAPI)
-
-						if err != nil {
-							// errors from getMovieData
-							errors = append(errors, err.Error())
-						}
-
-					} else {
-						// Errors from sourceAPI.getTitle
-						errors = append(errors, err.Error())
-					}
-				} else {
-					errors = append(errors, "Tmdb API usage was not enabled by the site administrator")
-					rerenderSite = true
-					return nil, errors, rerenderSite
+		if err != nil {
+			return nil, nil
```

Comment:
The error from handleTmdb should be logged here.

Reply:
Will be changed in the next commit